src/java.desktop/share/classes/javax/swing/plaf/basic/BasicSliderUI.java

```diff
@@ -150,6 +150,8 @@ public class BasicSliderUI extends SliderUI{
 
     /**
      * Constructs a {@code BasicSliderUI}.
+     *
+     * @since 16
```

Comment:
Hmm, the *explicit* default constructor was added in JDK 16, but it was implicit before then.
So I am not 100% sure what the right answer is - the same as the class, or when it was explicitly added.

Reply:
When mapping methods and when they first appeared (by using the historical record built into javac) I use an id in the form of 
`method: <erased-return-descriptor> <binary-name-of-enclosing-class>.<method-name>(<ParameterDescriptor>)` so for covariant overrides in general, when the return type changes I consider it to be a new method.

Looking at the contents of the dictionnary:
This explicit constructor existed for a long time but then this new was added a new one was added in JDK 16
| Key | Value |
| ------------- | ------------- |
| `method: void javax.swing.plaf.basic.BasicSliderUI.<init>(javax.swing.JSlider):` | 9 |
| `method: void javax.swing.plaf.basic.BasicSliderUI.<init>():`  | 16 |

Note: JDK 9 is used as the "base" as that's how far I can reliably use the `--release` info, so if something was added in JDK 2,5.7,9. It has a value of "9" in the dictionnary

Reply:
It seems that BasicSliderUI() was added by the mistake? it was not mentioned in the bug report...Seems it is too late to delete it?

Reply:
I'm sorry but `method: void javax.swing.plaf.basic.BasicSliderUI.<init>()` refers to the constructor, 
 as I use [this method](https://docs.oracle.com/en/java/javase/22/docs/api/java.compiler/javax/lang/model/element/ExecutableElement.html#getSimpleName()) to get a method's name.

I am saying that there was no default constructor before JDK 16 as it doesn't appear in the compiler's historical data until then and therefore warrants an `@since`

I am stealing my colleagues words but here is the general rule for when we want to add an `@since` until we publish a doc with rules for `@since`

> As a practical rule for deciding whether any declaration is new or not, imagine writing a test program that refers to the most specific form of the declaration. If that test program does not compile on JDK version N-1 and does compile on version N, then it warrants having `@since N`. Put another way, `@since N` should identify the first release in which the declaration can be used in the given form

Reply:
> It seems that BasicSliderUI() was added by the mistake? it was not mentioned in the bug report...Seems it is too late to delete it?

I agree. It shouldn't have been added.

Instead of adding `@since`, the constructor should be removed. It requires a CSR.

The longer it exists, the more chances there are that it's used.